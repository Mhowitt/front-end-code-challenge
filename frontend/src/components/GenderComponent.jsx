import React from 'react';
import { Link } from 'react-router-dom';

const GenderComponent = props => {
  return (
    <div>
      <h1>Hi {props.firstName}! It's great to have you!</h1>
      <h1>What's your gender?</h1>
      <button onClick={props.handleChange} name="gender" value='female'>Female</button>
      <button onClick={props.handleChange} name="gender" value='male'>Male</button>
      <div>
        <button><Link to="/profile/body-type"> Next</Link></button>
      </div>

    </div>
  )
}

export default GenderComponent;