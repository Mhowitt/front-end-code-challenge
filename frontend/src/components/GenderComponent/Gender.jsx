import React from 'react';
import { Link } from 'react-router-dom';
import './Gender.css';

const Gender = props => {
  return (
    <div>
      <h2>Hi {props.firstName}! It's great to have you! What's your gender?</h2>
      <div className="card-container">
        <div className="card">
          <div className="gender-container">
            <div className="gender-img female" />
            <Link to="/profile/body-type">
              <button
                onClick={props.handleClick}
                name="gender"
                value="female"
                className="gender-button"
              >
                I identify as a Female
              </button>
            </Link>
          </div>
          <div className="gender-container">
            <div className="gender-img male" />
            <Link to="/profile/body-type">
              <button
                onClick={props.handleClick}
                name="gender"
                value="male"
                className="gender-button"
              >
                I identify as a Male
              </button>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Gender;
