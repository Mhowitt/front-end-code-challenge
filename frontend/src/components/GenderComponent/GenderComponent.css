.gender-container {
  width: 228px;
  height: 400px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-self: center;
}

.gender-img {
  width: 228px;
  height: 400px;
  background-size: contain;
  background-repeat: no-repeat;
}

.female {
  background-image: url("../../woman.png");
}

.male {
  background-image: url("../../man.png");
}

.gender-button {
  align-self: flex-start;
  position: relative;
  width: auto;
  height: 40px;
  border-radius: 10px;
  background-color: #96858F;
  color: rgb(254, 252, 252);
  font-size: 14px;
  text-transform: uppercase;
  cursor: pointer;
  bottom: 0;
  opacity: 1;
  left: 5%;
  /* transition: bottom 1s linear; */
}
.gender-button:active {
  color:white;
}

/* .gender-container:hover .gender-button {
  bottom: 50%;
  opacity: 1;
} */

/* .text-button {
  opacity: 0;
  transition: opacity 1s ease-in;
} */

.card {
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 100%;
  justify-content: space-around;
}

/* .arrow {
  width: 50px;
  height: 50px;
  border-radius: 50%;
  border: 2px solid #96858F;
  transform: rotate(180deg);
  box-shadow: 0px 1px 20px -7px rgba(107,91,107,0.75);
  background-image: url('../../arrow.png');
  background-size: cover;
  cursor: pointer;
  transition: box-shadow 1.5s linear ease-out;
}

.arrow:hover {
  box-shadow: 0px 1px 25px -6px rgba(107,91,107,0.75);
}

.next-button {
  position: absolute;
  right: 1%;
  bottom: 0;
}

.card-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  position: relative;
} */
