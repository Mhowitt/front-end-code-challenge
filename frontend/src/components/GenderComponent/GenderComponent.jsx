import React from 'react';
import { Link } from 'react-router-dom';
import ProgressBar from '../ProgressBar/ProgressBar.jsx'
import './GenderComponent.css'

const GenderComponent = props => {
  return (
    <div>
      <h2>Hi {props.firstName}! It's great to have you! What's your gender?</h2>
      <ProgressBar question={props.question}/>
      <div className="card-container">
        <div className="card">
          <div className="gender-container">
            <div className="gender-img female"></div>
            <button onClick={props.handleChange} name="gender" value='female' className="gender-button">
            <span className="text-button">I identify as a Female</span></button>
          </div>
          <div className="gender-container">
            <div className="gender-img male"></div>
            <button onClick={props.handleChange} name="gender" value='male' className="gender-button">
            <span className="text-button">I identify as a Male</span></button>
          </div>
        </div>
        <div className="next-button">

          <Link to="/profile/body-type"><div className="arrow"></div></Link>
        </div>
        </div>
      </div>
  )
}

export default GenderComponent;