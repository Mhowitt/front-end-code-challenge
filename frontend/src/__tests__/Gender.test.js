import React from 'react'; // for JSX
import { shallow, mount } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import GenderComponent from '../components/GenderComponent/Gender.jsx';

describe('<GenderComponent />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = shallow(<GenderComponent />);
  });
  it('matches snapshot', () => {
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
