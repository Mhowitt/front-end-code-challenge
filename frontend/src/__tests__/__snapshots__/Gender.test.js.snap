// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenderComponent /> matches snapshot 1`] = `
<div>
  <h2>
    Hi 
    ! It's great to have you! What's your gender?
  </h2>
  <div
    className="card-container"
  >
    <div
      className="card"
    >
      <div
        className="gender-container"
      >
        <div
          className="gender-img female"
        />
        <Link
          replace={false}
          to="/profile/body-type"
        >
          <button
            className="gender-button"
            name="gender"
            value="female"
          >
            I identify as a Female
          </button>
        </Link>
      </div>
      <div
        className="gender-container"
      >
        <div
          className="gender-img male"
        />
        <Link
          replace={false}
          to="/profile/body-type"
        >
          <button
            className="gender-button"
            name="gender"
            value="male"
          >
            I identify as a Male
          </button>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
